### DISCUSSION TITLE
Utilisation de VIcon dans le cadre d’une CSP stricte

### DISCUSSION THREAD

**Discussion by CopperGiraffe** [dnum-mi/vue-dsfr#969] [Ideas]
Hello l’équipe vue-dsfr,

L’usage de VIcon et derrière de Iconify n’est pas toujours possible en fonction des environnements.
Certains projets auprès de certains ministères imposent une CSP stricte qui ne permet pas d’aller chercher les SVG vers l’extérieur. Également lorsque l’hébergement est géré côté ministère, il semble difficile d’obtenir une version self-hosted de Iconify.

Il faudrait toujours privilégier la solution proposé par le DSFR (notamment via ses classes CSS) pour ajouter des icones, le `VIcon` peut toujours exister, simplement il devrait pouvoir fonctionner dans un environnement "isolé" (en tout cas pour les icones présentent dans le dsfr)

**Comment by laruiss** [comment]
Bonjour @CopperGiraffe,

C’est tout à fait vrai, et je fais des essais avec unocss et je rencontre qqs problèmes. Si vous avez des suggestions, parlez-en nous !

**Reply by laruiss** [reply]
Pour cela, veuillez consulter la [partie dédiée de la documentation](https://vue-ds.fr/guide/icones#eviter-les-appels-reseaux-optionnel-pour-les-applications-internes)
